### ISSUE TITLE
MTG Forge - script to install App from official ftp server

### ISSUE THREAD

**Issue by Archargelod** [Botspot/pi-apps#2264]
### What is the name of the app?

Forge

### (Optional) Where is the app hosted?

https://downloads.cardforge.org/dailysnapshots/
https://github.com/Card-Forge/forge

### About the app

Unofficial MTG Game/Rules Engine
Runs on Java VM, so this game supports both 32 and 64-bit architectures.
Snapshot version, because releases are quickly outdated. Snapshots aren't automatic, they're tested before uploads.

### Upload file or Add PR Link

[Forge.zip](https://github.com/Botspot/pi-apps/files/10764239/Forge.zip)

### Confirmations

- [X] I have confirmed that this app is legal and not piracy.
- [X] I have confirmed that this app has never been discussed in https://github.com/Botspot/pi-apps/issues and https://github.com/Botspot/pi-apps/pulls, and it is not in the Pi Apps app list.
- [X] I have confirmed that this app can run on Raspberry Pi.

**Comment by theofficialgman** [comment]
multiple issues with this app request
```
game_folder="$HOME/MTG-Forge"
snapshot_server="https://downloads.cardforge.org/dailysnapshots"
VERSION=$(curl "$snapshot_server/version.txt")
```
this downloads the latest snapshot at install time but the users install will be outdated once the next snapshot comes output. please move checking for a version to an auto update script and hardcode the version in the install script. the auto update will update the version as necessary.

```
#install_packages sed openjdk-17-jre || exit 1
```
why this this commented out? also openjdk-17 is not available on all OSs that we support. search pi-apps for temurin and you will find multiple scripts that use their jres and add the apt repo as necessary

```
#test for aria2, because wget can be very slow
if command -v aria2c; then
  aria2c -x 6 -k 1M -c "$snapshot_server/$archive_name" -d /tmp || error 'Failed to download latest snapshot!'
else
  wget -c "$snapshot_server/$archive_name" --directory-prefix /tmp || error 'Failed to download latest snapshot!'
fi
```
pi-apps automatically overrides wget in scripts with aria2c, please remove this

**Comment by theofficialgman** [comment]
```
game_folder="$HOME/MTG-Forge"
```
please avoid installing programs into the users home directory.

if it is desired that the user have write permissions into the program folder, then place it in `~/.local/share`, otherwise place it in root owned `/usr/local/share` and change the .desktop files into `/usr/local/share/applications` as well.

**Comment by theofficialgman** [comment]
> Fixed - I've set game_folder to "$HOME/.local/usr/share/mtg-forge" User should have write access to folder for modding purposes, for example - [forge/wiki/Create-Enemies](https://github.com/Card-Forge/forge/wiki/Create-Enemies)
> 
> > and change the .desktop files into /usr/local/share/applications as well.

no you misunderstood what I meant. only use the root owned /usr/local/share/applications if you place the app in /usr/local/share. your previous locations for the .desktop files were correct with the gamefiles in ~/.local/share


also, and this is detrimental to inclusion in pi-apps, https://downloads.cardforge.org/dailysnapshots/ does not include previous days snapshots. so this app will break every day. github releases for this app also only include one version, older versions are deleted. this is not suitable for pi-apps, we need old versions to be cached as we only update apps weekly and users are not expected to always have the latest app versions on their local pi-apps installs. 

I suggest you create an issue upstream to solve one or both of those issues. for now this is blocked.

**Comment by Archargelod** [comment]
> github releases for this app also only include one version, older versions are deleted.
I don't believe this is true, only release right now is actually the very first release of Forge on Github. Earlier versions were published on "slightly-magic.net" forums. But releases are rare, maybe twice a year? 

> I suggest you create an issue upstream
Here's the [Issue](https://github.com/Card-Forge/forge/issues/2501).

In the meantime I made this repo with a simple workflow, that downloads snapshots daily and publishes them as releases.
And I rewrote install/update scripts to use it. **Is it too jank for pi-apps?** =)

[Forge.zip](https://github.com/Botspot/pi-apps/files/10776561/Forge.zip)

**Comment by Archargelod** [comment]
Sorry for the delay. It seems Forge won't make previous snapshots available. 
But, at least, releases are gonna be, well, released more often. That, I think, should be enough for most users. 

I updated script to install stable version of forge from official repo releases page.
I also added a simple cleanup trap to the install script.

[Forge.zip](https://github.com/Botspot/pi-apps/files/10888915/Forge.zip)

**Comment by theofficialgman** [comment]
> In the meantime I made this repo - [forge-snapshot-archive](https://github.com/Archargelod/forge-snapshot-archive), with a simple workflow, that downloads snapshots daily and publishes them as releases. And I rewrote install/update scripts to use it. **Or, it is too janky for pi-apps?** = )

sorry for the long time without a response. I saw your issue you made upstream and appreciate the dialog you had there.

it looks like upstream is going to do better about publishing stable releases to their github so I would prefer that those be used. while your forge-sanpshot-archive repo could be used, yes its a bit too janky for me to include since neither pi-apps nor the app maintainer controls it. I think the daily snapshots might be too unstable for general users on pi-apps anyway.

**Comment by theofficialgman** [comment]
@Archargelod upstream has been publishing to their github releases for a few stable releases now. I think those can be used.

also, installing java-17 has been greatly simplified with new functions added to pi-apps api
refer to the other pi-apps scripts for what to do 

https://github.com/Botspot/pi-apps/blob/97bcb5208a9d6bfd1b9f27a701e51ea3bfa74e7f/apps/Minecraft%20Java%20MultiMC5/install#L22-L47

also upstream needs to correct their `.sh` files to respect the `JAVA_HOME` variable if it is present. they directly call `java` which will be whatever the last version of java that is installed is.

**Comment by theofficialgman** [comment]
Please see the above comment and update the zip with the requested changes

**Comment by Archargelod** [comment]
@theofficialgman
It should be more convenient to use pull requests, so we can move discussion here: #2640
And here is the upstream PR for "JAVA_HOME": Card-Forge/forge/pull/6008
